protocol/kafkaapi/api_versions_request.go

```diff
@@ -0,0 +1,46 @@
+package kafkaapi
+
+import (
+	"fmt"
+
+	"github.com/codecrafters-io/kafka-tester/protocol/encoder"
+	"github.com/codecrafters-io/kafka-tester/protocol/kafkaapi/headers"
+)
+
+type ApiVersionsRequestBody struct {
+	// Version defines the protocol version to use for encode and decode
+	Version int16
+	// ClientSoftwareName contains the name of the client.
+	ClientSoftwareName string
+	// ClientSoftwareVersion contains the version of the client.
+	ClientSoftwareVersion string
+}
+
+func (r ApiVersionsRequestBody) encode(enc *encoder.Encoder) {
+
+	if r.Version < 3 {
+		panic(fmt.Sprintf("CodeCrafters Internal Error: Unsupported API version: %d", r.Version))
+	}
+	enc.WriteCompactString(r.ClientSoftwareName)
+	enc.WriteCompactString(r.ClientSoftwareVersion)
+	enc.WriteEmptyTagBuffer()
+}
+
+func (r ApiVersionsRequestBody) Encode() []byte {
```

Comment:
Let's merge `Encode` and `encode`, don't need the split, doesn't really make it any more readable